### ISSUE TITLE
Multiple instances under multiple user setup cause ~20% CPU usage

### ISSUE THREAD

**Issue by githubyouser** [emoacht/Monitorian#651]
My computer seems to be running a lot "hotter" recently, with the fan running almost constantly. I checked Task Manager and realized that Monitorian just seems to be sitting there eating CPU cycles. 

![image](https://github.com/user-attachments/assets/033f0911-a0ad-476e-a815-b78aed65b846)

It jumps up and down, but Monitorian uses between 20-30% of CPU constantly. I've been watching it for several minutes now, and it hasn't slowed down. I finally killed the process, because it's kinda problematic. Interestingly, if I then restart Monitorian it settles down to 0% CPU usage after the initial spike for checking monitors. I suspect that sleeping/resuming is what causes the runaway CPU usage. 

I've also been experiencing mouse delays/stutters at login and never made the connection, but after searching through the issues here for performance issues, I see that that's also something that's been reported, so Monitorian might be relevant to that issue as well. 

If you need any additional info, I'll be happy to supply it, please just let me know. Thank you for making this software available for free! I use it all the time and 99% of the time I never even think about it, it just works. :) 

Specs: 
Windows 11
Monitorian 4.9.1.0
ASUS TUF laptop
CPU: AMD Ryzen 5 3550H with Radeon Vega Mobile Gfx
Acer SA242Y external monitor (HDMI)

**Comment by emoacht** [comment]
Thank you for reporting.
I think I have never experienced such high CPU usage.
I have no idea but operation.log may help. How to produce the log is explained in [Reporting](https://github.com/emoacht/Monitorian#reporting) section of readme.

**Comment by MagicAndre1981** [comment]
Use ETW to capture a trace with wpr.exe from an elevated command prompt (**wpr -start CPU** ENTER wait until usage dropped and run `wpr -stop %USERPROFILE%\MonitorianPlusCPUUsage.etl -compress` ENTER)  and [analyze it](https://learn.microsoft.com/en-us/windows-hardware/test/wpt/cpu-analysis#cpu-usage-sampled-graph) with [Windows Performance Analyzer ](https://www.microsoft.com/store/productId/9N0W1B2BXGNZ)([WPA.exe](https://web.archive.org/web/20200919152140/https://channel9.msdn.com/Shows/Defrag-Tools/Defrag-Tools-42-WPT-CPU-Analysis#time=10m09s)).

**Comment by githubyouser** [comment]
> Thank you for reporting. I think I have never experienced such high CPU usage. I have no idea but operation.log may help. How to produce the log is explained in [Reporting](https://github.com/emoacht/Monitorian#reporting) section of readme.

Thanks, I've enabled Reporting, but I haven't seen the issue again yet. I've seen it at least twice in the past week or so, so hopefully I'll be able to catch it next time it happens.

**Comment by githubyouser** [comment]
So I left my laptop running and stepped away to eat supper (probably about an hour). When I returned, I could hear the CPU fans running as soon as I stepped into the room. The laptop and HDMI monitor were both sleeping and awoke as soon as I moved the mouse. I checked Task Manager, and sure enough Monitorian is using the CPU constantly again. Here's a copy of the [operation.log](https://github.com/user-attachments/files/17927678/operation.log)

**Comment by githubyouser** [comment]
> Use ETW to capture a trace with wpr.exe from an elevated command prompt (**wpr -start CPU** ENTER wait until usage dropped and run `wpr -stop %USERPROFILE%\MonitorianPlusCPUUsage.etl -compress` ENTER) and [analyze it](https://learn.microsoft.com/en-us/windows-hardware/test/wpt/cpu-analysis#cpu-usage-sampled-graph) with [Windows Performance Analyzer ](https://www.microsoft.com/store/productId/9N0W1B2BXGNZ)([WPA.exe](https://web.archive.org/web/20200919152140/https://channel9.msdn.com/Shows/Defrag-Tools/Defrag-Tools-42-WPT-CPU-Analysis#time=10m09s)).

Thanks, @MagicAndre1981 I tried running this for a while and then stopping it. It says "the trace was successfully saved" but no .etl file can be found in the directory. :( Not sure what to do about that.

**Comment by emoacht** [comment]
Thank you for the log.
I found the OS constantly fired display settings changed event.
So this issue happens when your PC suspends?

**Comment by MagicAndre1981** [comment]
if you found the ETL file does it show anything usefull in callstack of monitorian?

**Comment by githubyouser** [comment]
> if you found the ETL file does it show anything usefull in callstack of monitorian?

No ETL file is created. I tried twice. Not sure what else to try.

**Comment by MagicAndre1981** [comment]
hm, wired. You can replace `%USERPROFILE%` with a fixed path like `C:\ETLTraces\` for example.

**Comment by emoacht** [comment]
@githubyouser So does it happen when the display is turned off but the system has not suspended yet?

**Comment by githubyouser** [comment]
> @githubyouser So does it happen when the display is turned off but the system has not suspended yet?

Yes, that's correct. It just happened yesterday again, but I didn't have a chance to do anything about it because I was busy with other things.

**Comment by emoacht** [comment]
@githubyouser Thank you for the onfirmation.
Ant additional information?

**Comment by emoacht** [comment]
@githubyouser Thank you for the confirmation.
Ant additional information?

**Comment by githubyouser** [comment]
Hi @emoacht I haven't noticed it again till today. Then when I logged in to my other user account on the PC, I noticed almost right away that the fans were running hard and sure enough, Monitorian is chewing up ~30% of the CPU again. I tried running the trace again and saving to a different location as @MagicAndre1981 suggested, and this time I got an ETL file! It's 750MB. I took a look at it, but I don't really know what to look for. Any ideas? 
![image](https://github.com/user-attachments/assets/c4cdf8d4-7ca2-41d7-8f7b-ea3e1e15d7ed)

**Comment by emoacht** [comment]
I am not an expert of that tool.
If you can identify names of methods that were called during the period, it might be helpful to understand what is happening.

**Comment by MagicAndre1981** [comment]
load the [debug symbols (PDBs)](https://learn.microsoft.com/en-us/windows-hardware/test/wpt/load-symbols-or-configure-symbol-paths) and expand the full stack of minitorian to see what is called. On the point where the % Weight splitts you have the causing function call.

**Comment by githubyouser** [comment]
I just realized that I've been chasing a red herring. The cause of Monitorian's CPU usage is not sleeping, but switching between user accounts in Windows! I have 2 non admin user accounts on my PC and I regularly switch between them. (They both run Monitorian on login). If I don't log out but just switch to the other account, then the CPU usage immediately starts from Monitorian. It was camouflaged before, because logging in usually causes a bunch of CPU usage as various startup programs launch. So I wouldn't notice it till the PC was idle, and then realize that the CPU fans were _still_ going. 

I can reliably trigger the bug now by switching between user accounts without logging out of the first one.

**Comment by emoacht** [comment]
Thank you gor update.
Could you upload operation.log as is?
I have not seen the clue of user session change  in previously uploaded.

**Comment by githubyouser** [comment]
[operation.log](https://github.com/user-attachments/files/18118922/operation.log)
Sure, here's a copy of the latest operation.log. I had killed the Monitorian process, so I had to start it again, and as soon as I started it, it went back to using ~30% of the CPU.

**Comment by emoacht** [comment]
Thank you for the log.
Could you disable "restore brightness on reconnection" and see if anything changes?

**Comment by githubyouser** [comment]
> Could you disable "restore brightness on reconnection" and see if anything changes?

I disabled it in both user profiles but the issue still shows up. When I logged out from one profile, Monitorian's CPU usage immediately went back down to normal. So it only happens when both profiles are logged in and running Monitorian at the same time.

**Comment by emoacht** [comment]
I found that a preceding instance of this app prevents a succeeding instance from starting named pipe server and causes unauthorized access exception. Consequently, repetitive attemps to start named pipe server caused relatively high CPU usage. It only happens under multiple user setup.

I changed to stop the attempt if that exception is thown. See 3b154cce2adc5e91de317c883531e08e4c439ccd
A succeeding instance will run without CPU usage but it will no longer accept command-line option to get/set brightness/constrast.
